code/modules/power/apc/apc_main.dm

```diff
@@ -482,6 +487,11 @@
 		force_update = TRUE
 		return
 
+	if((obj_flags & EMAGGED || malfai) && icon_state == "apc0") //apc0 means we're not broken and the cover's not open.
```

Comment:
Checking icon state directly for code comparisons is very.. iffy

Can we not just check for broken or opened? In fact, don't we already do that a bit further up the proc?

Reply:
My reasoning for it was more that we *only* want the overlay to show if the `apc0` sprite is showing, because no matter what, if the screen isn't visible, the overlay shouldn't be either.

That said, you're right, there are already checks for broken and opened earlier, so the check here doesn't mean much.